application-tour-ui/src/main/resources/TourCode/StepSheet.xml

```diff
@@ -0,0 +1,537 @@
+   });
+
+   // Initialize save step button
+   $( "body" ).on( "click", "#saveStepBtn", function() {
+      var parent = $(this).parent().parent();
+      var stepOrder = $("#stepOrder", parent).val();
+      //var prefix = "TourCode.StepClass_" + (stepOrder ? stepOrder:"0") + "_";
+      var prefix = "TourCode.StepClass_";
+      var BSTDocRef = $("#BSTDocRef",parent).val();
+      var element = $("input[id^='" + prefix + "'][id$='_element']", parent).val();
+      var title = $("input[id^='" + prefix + "'][id$='_title']", content).val();
+      var backdrop = $("input[id^='" + prefix + "'][id$='_backdrop']", parent).is(":checked");
+      var content = $("textarea[id^='" + prefix + "'][id$='_content']", parent).val();
+      var placement = $("select[id^='" + prefix + "'][id$='_placement']", parent).val();
+      var targetPage = $("input[id^='" + prefix + "'][id$='_targetPage']", parent).val(); 
+      var nbSteps = $("#nbsteps").val();
+      var step = {element: (element? element: ""),title: (title? title: ""), backdrop: (backdrop? backdrop :0), content: (content? content: ""), placement:(placement? placement: ""), targetPage:(targetPage? targetPage: ""), BSTDocRef: (BSTDocRef? BSTDocRef: ""), nbSteps:nbSteps, stepOrder:stepOrder, action:"editStep", xpage:"plain"};
+      $.ajax({
+	 url:"$xwiki.getURL('TourCode.AjaxActions')",
+	 type:'POST',
+	 data: step, 
+	 success: function(rep, textStatus, jqXHR) {
+            $('#stepsContainer').html(rep);
+         },
+         complete:function( jqXHR , textStatus ) {
+            // Close dialog box
+            $(".xdialog-close").click(); 
+         }
+      });
+   });
+
+   // Initialize step edit btns
+   $( "body" ).on( "click", ".editStep", function() {
+      $('.xdialog-modal-container').remove();
+      var BSTDocRef = $("#BSTDocRef").val();
+      var stepOrder = $(this).attr("rel");
+      new XWiki.widgets.StepPopup({pageURL: bstStepURL + '&amp;BSTDocRef=' + encodeURIComponent(BSTDocRef) + '&amp;stepOrder=' + encodeURIComponent(stepOrder)});            
+   });
+   
+   // Initialize step delete btns
+   $( "body" ).on( "click", ".deleteStep", function() {
+      $('.xdialog-modal-container').remove();
+      var BSTDocRef = $("#BSTDocRef").val();
+      var stepOrder = $(this).attr("rel");
+      new XWiki.widgets.ConfirmationBox({onYes: function() {
+         new XWiki.widgets.Notification("Deleting step ...", 'inprogress');
+         $.ajax({
+	    url: actionsURL + "&amp;action=deleteStep" + "&amp;BSTDocRef=" + encodeURIComponent(BSTDocRef) + "&amp;stepOrder=" + encodeURIComponent(stepOrder),
+            type: 'POST',
+            success: function(rep, textStatus, jqXHR) {
+               $('#stepsContainer').html(rep);
+            },
+	    complete: function(event, XMLHttpRequest, ajaxOptions) {
+               $(".xnotification-inprogress").remove();
+               new XWiki.widgets.Notification("$services.localization.render('Delete step done!')","done");
```

Comment:
For safety reasons, please use $escapetool.javascript($services.localization.render('...')).